python/lsst/ip/isr/ampOffset.py

```diff
@@ -144,10 +149,28 @@ def run(self, exposure):
         ampDims = [amp.getBBox().getDimensions() for amp in amps]
         if not all(dim == ampDims[0] for dim in ampDims):
             raise RuntimeError("All amps should have the same geometry.")
+        else:
+            # The zeroth amp is representative of all amps in the detector.
+            self.ampDims = ampDims[0]
+            # Dictionary mapping side numbers to interface lengths.
+            # See `getAmpAssociations()` for details about sides.
+            self.interfaceLengthLookupBySide = {i: self.ampDims[i % 2] for i in range(4)}
+
+        # Determine amplifier geometry.
+        # ampWidths = {amp.getBBox().getWidth() for amp in amps}
+        # ampHeights = {amp.getBBox().getHeight() for amp in amps}
+        ampAreas = {amp.getBBox().getArea() for amp in amps}
+        if len(ampAreas) > 1:
+            raise NotImplementedError(
+                "Amp offset correction is not yet implemented for detectors with differing amp sizes."
+            )
 
         # Assuming all the amps have the same geometry.
         self.shortAmpSide = np.min(ampDims[0])
 
+        # TODO: Double check that this is true.
```

Comment:
Has this been done? If not, this needs to be assigned to a Jira ticket to capture future work.